src/context/WalletProvider/XDEFI/components/Connect.tsx

```diff
@@ -58,10 +56,8 @@ export const XDEFIConnect = ({ history }: XDEFISetupProps) => {
           throw new Error('walletProvider.xdefi.errors.multipleWallets')
         }
 
-        // Switch to Mainnet if wallet is on any other chain
-        const chainId = await wallet.ethGetChainId?.()
-        if (bnOrZero(chainId).toString() !== CHAIN_REFERENCE.EthereumMainnet) {
-          await wallet.ethSwitchChain?.(bn(CHAIN_REFERENCE.EthereumMainnet).toNumber())
+        if (provider?.chainId !== 1) {
```

Comment:
Don't need the full revert here, cherry-pick hunks of https://github.com/shapeshift/web/commit/f304b4f416a99472525067a96240179fe1cc4855 to still use ethGetChainId(), but not the chain switch part. Also remove the wrong network error, we should not need to error anymore or wrong EVM chain since we can just switch it before broadcasting a Tx